### PR TITLE
UserPrefs - Added Configurable Hops

### PR DESCRIPTION
**Pull request by medentem**
- Added configurable hops to UserPrefs

### CODE REVIEW

File: src/mesh/NodeDB.cpp [hunk 1]
```diff
@@ -308,7 +308,11 @@ void NodeDB::installDefaultConfig(bool preserveKey = false)
 #else
     config.lora.modem_preset = meshtastic_Config_LoRaConfig_ModemPreset_LONG_FAST;
 #endif
+#ifdef USERPREFS_CONFIG_LORA_HOPS
+    config.lora.hop_limit = USERPREFS_CONFIG_LORA_HOPS;
+#else
```
Comment (caveman99) [inline src/mesh/NodeDB.cpp:313]: Given the history we've had with this setting, i am strongly not a fan of making it bakeable into firmware.

Comment (medentem) [reply src/mesh/NodeDB.cpp:313]: It's use case dependent, and is currently manipulated via the`USERPREFS_EVENT_MODE` user prefs for valid reasons. What is the specific concern?

**Comment by medentem** [comment]
Closing to rebase and clean up PR.

File: src/mesh/NodeDB.cpp (hunk 1 above)
Comment (caveman99) [reply src/mesh/NodeDB.cpp:313]: People giving out compiled firmware with this set to "7" and marketing it as "meshtastic range boosted special hacker version"

Comment (thebentern) [reply src/mesh/NodeDB.cpp:313]: I'm not in favor of this either. We're working on making an interface to allow users to custom define userPrefs for custom builds. Allowing users to bake in 7 hop default firmware is like handing out the plans on how to build a mesh killing gun.

**Pull request closed without merging**